File: gitdot-auth/src/handler/logout.rs
use axum::{Json, extract::State, http::StatusCode};

use gitdot_api::endpoint::auth::logout as api;
use gitdot_core::dto::LogoutRequest;

use crate::app::{AppError, AppResponse, AppState};

pub async fn logout(
    State(state): State<AppState>,
    Json(body): Json<api::LogoutRequest>,
) -> Result<AppResponse<()>, AppError> {
    let request = LogoutRequest {
        refresh_token: body.refresh_token,
    };
    state
        .session_service
        .logout(request)
        .await
        .map_err(AppError::from)
        .map(|_| AppResponse::new(StatusCode::NO_CONTENT, ()))
}

File: gitdot-web/packages/gitdot-api/src/endpoint/auth/logout.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const LogoutRequest = z.object({
  refresh_token: z.string(),
});
export type LogoutRequest = z.infer<typeof LogoutRequest>;

export const Logout = {
  path: "/auth/logout",
  method: "POST",
  request: LogoutRequest,
  response: z.void(),
} as const satisfies Endpoint;

export type Logout = typeof Logout;
